web/src/api/auth/login-callback.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  getSession,
+  updateSession,
+  withApiAuthRequired,
+} from "@auth0/nextjs-auth0";
+import { NextApiRequest, NextApiResponse } from "next";
+import { errorResponse } from "src/backend/errors";
+
+import {
+  FetchUserQuery,
+  getSdk as fetchUserSdk,
+} from "./graphql/fetch-user.generated";
+
+import { getAPIServiceGraphqlClient } from "src/backend/graphql";
+import { urls } from "src/lib/urls";
+import { getSdk as addAuth0Sdk } from "./graphql/add-auth0.generated";
+import { Auth0User } from "src/lib/types";
+import { isEmailUser } from "src/lib/utils";
+
+const wait = (time: number) =>
+  new Promise((resolve) => setTimeout(resolve, time));
+
+export const auth0Login = withApiAuthRequired(
+  async (req: NextApiRequest, res: NextApiResponse) => {
+    const session = await getSession(req, res);
+
+    if (!session) {
+      return errorResponse(
+        res,
+        500,
+        "internal_server_error",
+        "Something went wrong",
+        null,
+        req
+      );
+    }
+
+    const client = await getAPIServiceGraphqlClient();
+    const auth0User = session.user as Auth0User;
+    let user: FetchUserQuery["user"][number] | null | undefined = null;
+
+    if (isEmailUser(auth0User) && !auth0User.email_verified) {
+      return res.redirect(307, urls.logout({ error: true }));
+    }
+
+    if (isEmailUser(auth0User)) {
```

Comment:
logic here should be to check first whether the user has an auth0Id in the db and that takes precedence, otherwise check the logic for email / sign in with worldcoin nullifier

Reply:
related to this, we should be updating the user's email address / name if it changed (can be in a follow up PR)